src/main/scala/PropertyChecker.scala

```diff
@@ -0,0 +1,408 @@
+          None
+        } else {
+          Option(Right(objectNode.get("maxInclusive").get("dateTime").asText()))
+        }
+      }
+
+      val minExclusive:Option[Either[Int, String]] = if (objectNode.path("minExclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("minExclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("minExclusive").asInt()))
+        }
+      } else {
+        if (objectNode.get("minExclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("minExclusive").get("dateTime").asText()))
+        }
+      }
+
+      val maxExclusive:Option[Either[Int, String]] = if (objectNode.path("maxExclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("maxExclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("maxExclusive").asInt()))
+        }
+      } else {
+        if (objectNode.path("maxExclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("maxExclusive").get("dateTime").asText()))
+        }
+      }
+
+     (minInclusive, minExclusive, maxInclusive, maxExclusive) match {
+       case (Some(Left(minI)), Some(Left(minE)), _, _) => throw new MetadataError(s"datatype cannot specify both minimum/minInclusive ($minI) and minExclusive ($minE)")
+       case (Some(Right(minI)), Some(Right(minE)), _, _) => throw new MetadataError(s"datatype cannot specify both minimum/minInclusive ($minI) and minExclusive ($minE)")
+       case (_, _, Some(Left(maxI)), Some(Left(maxE))) => throw new MetadataError(s"datatype cannot specify both maximum/maxInclusive ($maxI) and maxExclusive ($maxE)")
+       case (_, _, Some(Right(maxI)), Some(Right(maxE))) => throw new MetadataError(s"datatype cannot specify both maximum/maxInclusive ($maxI) and maxExclusive ($maxE)")
+       case (Some(Left(minI)), _,Some(Left(maxI)), _) if minI > maxI => throw new MetadataError(s"datatype minInclusive ($minI) cannot be more than maxInclusive ($maxI)")
+       case (Some(Right(minI)), _,Some(Right(maxI)), _) if minI.asInstanceOf[Int] > maxI.asInstanceOf[Int] => throw new MetadataError(s"datatype minInclusive ($minI) cannot be more than maxInclusive ($maxI)")
+       case (Some(Left(minI)), _, _, Some(Left(maxE))) if minI >= maxE => throw new MetadataError(s"datatype minInclusive ($minI) cannot be greater than or equal to maxExclusive ($maxE)")
+       case (Some(Right(minI)), _, _, Some(Right(maxE))) if minI.asInstanceOf[Int] >= maxE.asInstanceOf[Int] => throw new MetadataError(s"datatype minInclusive ($minI) cannot be greater than or equal to maxExclusive ($maxE)")
+       case (_, Some(Left(minE)), _, Some(Left(maxE))) if minE > maxE => throw new MetadataError(s"datatype minExclusive ($minE) cannot be greater than or equal to maxExclusive ($maxE)")
```

Comment:
You might be able to summarise both these comparisons as `Some(Either(minE)), _, Some(Either(maxE))) if minE > maxE`. See what the compiler says.